### ISSUE TITLE
Fix various false positives/negatives after `@property` spec changes

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#9468]
[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAHAThFM0BcCeABALREBmEEANMUQEYCGaBwAOgHYEEDOebO9AD3gEAPJAA2ENAD4A3OwC+7dvAwQczdpxLkIwgIwoB8tkrbtUGLLkIkGAL00cCASzYALbC5xdhONACuMCacbt4u9OJEAG6RQcJoMFAh3Lz8QgQsIGIQkjJZJmbs7jgAtuJO2nT09gZGhSCUIKQu4jAAcvSlcIgwAl0obQB0YFxcjeAQbC0A5gggfShSGrCk9AHiGqzOfTgwbFC+BADaAORc+G3ibjhEkNMuM0QX9AeMUKcAupSKJhP3swAYlJSvQcPMLngrjcRlNZkNSgArcZNWAocaIbacLKQ6F8LLCLLiMEwC5ZH7OHGXGDXPh3OGPZ78N5oKAEzIgYl7MkgRTsEAKIA)

This is intended to be a tracking issue with more detailed and specific issues to be filed for individual problems.

https://drafts.css-houdini.org/css-properties-values-api/#at-property-rule

These changes where made to the spec or resolved to be made:
- the prelude now allows a list of property names
- descriptors are now optional
- [the `syntax` descriptor now also accepts `<syntax>` instead of only `<string>`](https://github.com/w3c/csswg-drafts/issues/11426)

I don't currently have a lot of time, but still wanted to give a heads-up for these additions as I think some of the planned updates build on `@property`?

**Comment by jeddy3** [comment]
Thanks for the heads-up!

> I think some of the planned updates build on `@property`?

Yes, we'll be collecting custom properties and their values for subsitition in these updates. We should accept lists of property names.

> the prelude now allows a list of property names

We'll need to apply the same fix to `no-unknown-custom-properties`, too: [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAHAThFM0BcCeABALREBmEEANMUQEYCGaBwAOgHYEEDOebO9AD3gEAPJAA2ENAD4A3O04BLNopyL64ogDcNAVxjC0MKPLYBfdu3rMFBCVOE60AChLkIASlNmQlEKUVxGAA5egBbOEQYAXCUIIA6MC4uX3AINgCAcwQQaJQpHAJYUnpdcULWDgIWEDQymC4a4UrOThq2CCJdNgBrDoB3NiIwXS4cCDCidExsNQamghw6mFsLc1NUyAzFTIAxKTD6HByxvCDxZRxE9Kz4sIArFL9YFBTEFuqQU-PLhZrxI4NHA1Si2GrfGAXPjDG47IhjehsKCMKB-EAAnBAmrsNYgMxAA)

A CSSTree patch will hopefully fix the [`at-rule-prelude-no-invalid`](https://stylelint.io/user-guide/rules/at-rule-prelude-no-invalid) false positive.

> descriptors are now optional

Seemingly only impacts an upcoming rule:
- https://github.com/stylelint/stylelint/issues/8212

Tracking there.

> `syntax` descriptor accepts `<syntax>`

CSSTree patch again for [`at-rule-descriptor-value-no-unknown`](https://stylelint.io/user-guide/rules/at-rule-descriptor-value-no-unknown) false positive.

The [`syntax-string-no-invalid`](https://stylelint.io/user-guide/rules/syntax-string-no-invalid) rule is still correct as it checks the `<string>` grammar.

Assuming CSSTree patches will fix the two false positives mentioned above, I'll open dedicated issues for the:
- `no-unknown-custom-properties` false positive
- `declaration-property-value-no-unknown` false negative

**Comment by romainmenke** [comment]
> >  the prelude now allows a list of property names

> A CSSTree patch will hopefully fix the [at-rule-prelude-no-invalid](https://stylelint.io/user-guide/rules/at-rule-prelude-no-invalid) false positive.

It did :)

The patches didn't include preludes yet, but now do: [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAHAThFM0BcCeABALREBmEEANMUQEYCGaBwAOgHYEECWbAFtlxwBneARxoArjADc7Tj0Fd6AGyIA3FVNFoYUWRwJC8bHPQAeoliAA8kZRDQA+K-oC+ISiFJdlMAHL0ALZwiDBmQSi+AHRgQkIe4BBs3gDmCCBhKA44BLCk9BLKOawGYTgwbFAiBADaAORC+L7KPDhEkMlcKUSN9JWMUHUAupTsrvoJHakAYg6B9DjpjXjNrTFJqVGBAFbxnrAo8YglnFbLqyZWliDKCzCNVqMGZ00wLSbtG109pv1oUFcCFZbuUHiAxuwQK4gA)

**Comment by jeddy3** [comment]
@romainmenke Thank you for patching the `@property` prelude.

> The patches didn't include preludes yet, but now do

This change may have triggered false positives for `@import`, e.g. both these are flagged:

```css
@import "foo.css";
@import url("bar.css");
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBLAtgBwgJwC4AIAdEAMwggDowBna4gbkIDtVMcCBXbAGwApiARgENsVWsQCU9EABpSKbjAByQtHEQwAHqoyKx1WeAhMSKAOYIQWrHnywSQjtwLBm+fFtwwmUavHwA2gDk1LgAnorcKEy4ALSQJuaxoUI+IlBBALoyzAC+jEyGCaZmAGI4aEK4lqERMFExVMYlFGgAVgZysBgGiK5M7sS1kdG4xP7E3FUwocQ5A0Qgw-Wj8c1JKWnYUOOLU16zIHnMILlAA).

See:
- https://github.com/stylelint/stylelint/issues/9472

Is this something to be resolved upstream, or should we patch Stylelint itself?

**Comment by romainmenke** [comment]
Thank you for the ping.

`@import` was one of the rules where I thought I wouldn't have to double check that the spec was actually correct 😅

Should be better now.
